components/org.wso2.carbon.identity.custom.header.filter/src/main/java/org/wso2/carbon/identity/custom/header/filter/CustomHeaderFilter.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.custom.header.filter;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.context.PrivilegedCarbonContext;
+import org.wso2.carbon.identity.configuration.mgt.core.exception.ConfigurationManagementException;
+import org.wso2.carbon.identity.configuration.mgt.core.model.Resource;
+import org.wso2.carbon.identity.custom.header.filter.function.ResourceToCustomHeader;
+import org.wso2.carbon.identity.custom.header.filter.internal.CustomHeaderDataHolder;
+import org.wso2.carbon.identity.custom.header.filter.model.Header;
+
+import java.io.IOException;
+import java.util.List;
+
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Custom Header Filter.
+ */
+public class CustomHeaderFilter implements Filter {
+
+    private static final Log log = LogFactory.getLog(CustomHeaderFilter.class);
+    private static final String RESOURCE_TYPE = "custom-headers";
+
+    @Override
+    public void init(FilterConfig filterConfig) {
+    }
```

Comment:
Can't we get the application name from filterConfig ?
filterConfig. getServletContext().getContextPath() should give the context path, so we we could be able to resolve the base URL without relying on the PrivilegedCarbonContext

Reply:
The idea here is to get the application name. Not the base URL. Do you know a way to get the application name?

Reply:
What happens when an application is renamed?